src/scripts/main.js

```diff
@@ -1,3 +1,131 @@
 'use strict';
 
 // write code here
+
+const table = document.getElementsByTagName('table');
+
+const people = table[0].children[1].children;
+
+table[0].children[0].children[0].children[0].addEventListener('click',
+  () => sortByName());
+
+table[0].children[0].children[0].children[1].addEventListener('click',
+  () => sortByPosition());
+
+table[0].children[0].children[0].children[2].addEventListener('click',
+  () => sortByAge());
+
+table[0].children[0].children[0].children[3].addEventListener('click',
+  () => sortBySalary());
+
+function sortByName() {
+  let switching = true;
+
+  let shouldSwitch;
+
+  let x, y, i;
+
+  while (switching) {
+    switching = false;
+
+    for (i = 0; i < (people.length - 1); i++) {
+      shouldSwitch = false;
+      x = people[i].children[0].innerText.toLowerCase();
+      y = people[i + 1].children[0].innerText.toLowerCase();
+
+      if (x.localeCompare(y) === 1) {
+        shouldSwitch = true;
+        break;
+      }
+    }
+
+    if (shouldSwitch) {
+      people[i].parentNode.insertBefore(people[i + 1], people[i]);
+      switching = true;
+    }
+  }
+}
+
+function sortByPosition() {
+  let switching = true;
+
+  let shouldSwitch;
+
+  let x, y, i;
+
+  while (switching) {
+    switching = false;
+
+    for (i = 0; i < (people.length - 1); i++) {
+      shouldSwitch = false;
+      x = people[i].children[1].innerText.toLowerCase();
+      y = people[i + 1].children[1].innerText.toLowerCase();
+
+      if (x.localeCompare(y) === 1) {
+        shouldSwitch = true;
+        break;
+      }
+    }
+
+    if (shouldSwitch) {
+      people[i].parentNode.insertBefore(people[i + 1], people[i]);
+      switching = true;
+    }
+  }
+}
+
+function sortBySalary() {
+  let switching = true;
+
+  let shouldSwitch;
+
+  let x, y, i;
+
+  while (switching) {
+    switching = false;
+
+    for (i = 0; i < (people.length - 1); i++) {
+      shouldSwitch = false;
+      x = +people[i].children[3].innerText.slice(1).replace(',', '');
+      y = +people[i + 1].children[3].innerText.slice(1).replace(',', '');
+
+      if (x > y) {
+        shouldSwitch = true;
+        break;
+      }
+    }
+
+    if (shouldSwitch) {
+      people[i].parentNode.insertBefore(people[i + 1], people[i]);
+      switching = true;
+    }
+  }
+}
+
+function sortByAge() {
```

Comment:
You are duplicating a lot of code in these functions
Instead of four functions create one that accepts the index of the column as an argument and sorts it accordingly, also, you may join some cases. (`switch case` block will be useful)

Also, you may transform the HTML collection to Array in this way: `[...people]` and use built-in sort function

Reply:
have rewritten the logic ) now it is only one main function and one call)